src/System.Text.Formatting/System/Text/Parsing/InvariantParser_long.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System.Diagnostics;
+using System.Text.Formatting;
+using System.Text.Utf8;
+
+namespace System.Text.Parsing
+{
+    public static partial class InvariantParser
+    {
+        /// <summary>
+        /// Parses an unsigned 64-bit integer from a location within a UTF-8 byte array buffer.
+        /// </summary>
+        /// <param name="utf8Text">The UTF-8 buffer, passed as a byte array.</param>
+        /// <param name="index">The index location of the value to be parsed within the buffer.</param>
+        /// <param name="value">The parsed value.</param>
+        /// <param name="bytesConsumed">The length (in bytes) of the unparsed value within the UTF-8 buffer.</param>
+        /// <returns>True if parsing is successful; false otherwise.</returns>
+		public static bool TryParse(byte[] utf8Text, int index, out long value, out int bytesConsumed)
+        {
+            Precondition.Require(utf8Text.Length > 0);
+
+            value = 0;
+            bytesConsumed = 0;
+            bool negative = false;
+
+            if (utf8Text[index] == '-')
+            {
+                negative = true;
+                index++;
+                bytesConsumed++;
+            }
+
+            for (int byteIndex = index; byteIndex < utf8Text.Length; byteIndex++) // loop through the byte array
+            {
+                if (utf8Text[byteIndex] < '0' || utf8Text[byteIndex] > '9') // if the next character is not a digit
+                {
+                    if (bytesConsumed == 0) // check to see if we've processed any digits at all
+                    {
+                        value = default(int); // if we haven't, set value to 0 and return false
+                        return false;
+                    }
+                    else
+                    {
+                        if (negative) // We check if the value is negative at the very end to save on comp time
+                        {
+                            value *= -1;
+                        }
+                        return true; // otherwise return true
+                    }
+                }
+                long candidate = (value * 10); // left shift the value
```

Comment:
Do we need long local here? I think it might be expensive on some architectures (32-bit architectures) to do these long operations and back and forth casts. Feel free to file an investigation issue for this and investigate later.